## 导入所需要的包

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import datetime
import xgboost   # 导入xgboost库
#from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import cross_val_score
#from feature_selector import FeatureSelector
#from sklearn.model_selection import LeaveOneOut
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import GridSearchCV
from xgboost import XGBClassifier  #导入sklearn分类API

## 加载数据集

In [3]:
X=pd.read_csv("11Mismatch21.csv",header=None)
y=pd.read_csv("label.csv",header=None)

## 数据预处理

In [4]:
data=pd.concat((X,y),axis=1)
data=np.array(data)
print(data.shape)
np.random.seed(611)
np.random.shuffle(data) 
data_X=data[:,0:-1]
data_y=data[:,-1]

(154, 17)


## 标准化数据

In [5]:
transfer=StandardScaler()
Standard_X=transfer.fit_transform(data_X)  #在该数据集中StandardScaler比MinMaxScaler用的好，数据整体精确度提高
Standard_Y=data_y
print(Standard_X.var())

1.0


## 建立模型并调参

In [6]:
from sklearn.model_selection import RepeatedStratifiedKFold
cv=RepeatedStratifiedKFold(n_splits=5,n_repeats=50,random_state=0) 

## 输出accuracy值

In [7]:
print(datetime.datetime.now())
estimator=XGBClassifier(n_estimators=80,learning_rate=0.22,max_depth=4,reg_lambda=0)
acc_val=cross_val_score(estimator,Standard_X,Standard_Y,scoring="accuracy",cv=cv)       #指标网址：https://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter
#print(acc_val)                                                                         #roc_auc
print(datetime.datetime.now())
print("the mean score of cross validation:{:.3f}".format(np.mean(acc_val)))

2022-02-10 02:29:35.437381
2022-02-10 02:29:41.211580
the mean score of cross validation:0.894


## 输出roc_auc

In [9]:
print(datetime.datetime.now())
estimator=XGBClassifier(n_estimators=80,learning_rate=0.22,max_depth=4,reg_lambda=0)
acc_val=cross_val_score(estimator,Standard_X,Standard_Y,scoring="roc_auc",cv=cv)       #指标网址：https://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter
#print(acc_val)                                                                         #roc_auc
print(datetime.datetime.now())
print("the mean score of cross validation:{:.3f}".format(np.mean(acc_val)))

2022-02-10 02:30:22.840067
2022-02-10 02:30:27.760896
the mean score of cross validation:0.949


## 输出f1

In [10]:
print(datetime.datetime.now())
estimator=XGBClassifier(n_estimators=80,learning_rate=0.22,max_depth=4,reg_lambda=0)
acc_val=cross_val_score(estimator,Standard_X,Standard_Y,scoring="f1",cv=cv)       #指标网址：https://scikit-learn.org/stable/modules/model_evaluation.html#scoring-parameter
#print(acc_val)                                                                         #roc_auc
print(datetime.datetime.now())
print("the mean score of cross validation:{:.3f}".format(np.mean(acc_val)))

2022-02-10 02:30:55.973000
2022-02-10 02:31:00.995557
the mean score of cross validation:0.893


## Accuracy:0.894    roc_auc=0.949    f1_score=0.893

## 最新一篇文章：ACC=0.8516    F1-score=0.8572    AUC=0.9117